Public/Remove-WinADDuplicateObject.ps1

```diff
@@ -24,8 +24,14 @@
     $Count = 0
     $DuplicateObjects = Get-WinADDuplicateObject @getWinADDuplicateObjectSplat
     foreach ($Duplicate in $DuplicateObjects | Select-Object -First $LimitProcessing) {
-        If($Duplicate.ProtectedFromAccidentalDeletion -eq "True"){
-            Get-ADObject -Filter {ObjectGUID -eq "$($Duplicate.ObjectGUID)"} -ErrorAction Stop | Set-ADObject -ProtectedFromAccidentalDeletion $false
+        If($Duplicate.ProtectedFromAccidentalDeletion -eq $true){
+            Try{
+                 Get-ADObject -Filter {ObjectGUID -eq "$($Duplicate.ObjectGUID)"} -ErrorAction Stop | Set-ADObject -ProtectedFromAccidentalDeletion $false -ErrorAction Stop
```

Comment:
Now that I think about it, Get-ADObject is not necessary. ObjectGUID is available as part of Duplicate.ObjectGUID, so is protected from accidental deletion so you can use Set-ADObject only.

Reply:
Yes, you are correct. removed Get-AdObject